Source/Lib/Fluxor.Blazor.Web.Analyzers/CallBaseOnInitialized.cs

```diff
@@ -23,91 +22,66 @@ public override void Initialize(AnalysisContext context)
+		{
 			return;
+		}
 
-		if (methodDecl.Modifiers.All(m => m.Text != "override"))
-			return;
+		if (!CallsBase(context.OperationBlocks, method.OverriddenMethod))
+			context.ReportDiagnostic(Diagnostic.Create(Rule, method.Locations[0]));
+	}
 
-		string methodName = methodDecl.Identifier.Text;
-		if (methodName != "OnInitialized" && methodName != "OnInitializedAsync")
-			return;
+	private bool CallsBase(ImmutableArray<IOperation> operations, IMethodSymbol? overriddenMethod)
+	{
+		foreach (var operation in operations)
+		{
+			if (CallsBase(operation, overriddenMethod))
+				return true;
+		}
 
-		SemanticModel semanticModel = context.SemanticModel;
-		INamedTypeSymbol? containingType = semanticModel.GetDeclaredSymbol(methodDecl)?.ContainingType;
-		if (containingType is null)
-			return;
+		return false;
+	}
 
-		if (!IsFluxorComponentBase(containingType))
-			return;
+	private bool CallsBase(IOperation operation, IMethodSymbol? overriddenMethod)
+	{
+		if (operation is IBlockOperation blockOperation)
+			return CallsBase(blockOperation.Operations, overriddenMethod);
 
-		bool callsBase =
-			methodDecl.Body is not null
-			&& methodDecl.Body.Statements
-				.OfType<ExpressionStatementSyntax>()
-				.Select(s => s.Expression)
-				.Select(expr =>
-					expr is AwaitExpressionSyntax awaitExpr
-						? awaitExpr.Expression
-						: expr)
-				.OfType<InvocationExpressionSyntax>()
-				.Any(inv => IsBaseCall(inv, methodName, semanticModel));
-
-		if (!callsBase && methodDecl.ExpressionBody is null)
-		{
-			context.ReportDiagnostic(Diagnostic.Create(Rule, methodDecl.Identifier.GetLocation()));
-			return;
-		}
+		if (operation is IExpressionStatementOperation expressionStatementOperation)
+			operation = expressionStatementOperation.Operation;
 
-		if (methodDecl.ExpressionBody is not null)
-		{
-			ExpressionSyntax expr = methodDecl.ExpressionBody.Expression;
-			if (
-				expr is InvocationExpressionSyntax exprInv
-				&& IsBaseCall(exprInv, methodName, semanticModel)
-			)
-			{
-				return;
-			}
-			context.ReportDiagnostic(Diagnostic.Create(Rule, methodDecl.Identifier.GetLocation()));
-		}
+		if (operation is IAwaitOperation awaitOperation)
+			operation = awaitOperation.Operation;
+
+		return operation is IInvocationOperation invocation &&
+			invocation.TargetMethod.Equals(overriddenMethod, SymbolEqualityComparer.Default);
 	}
 
-	private static bool IsBaseCall(
-		InvocationExpressionSyntax invocation,
-		string methodName,
-		SemanticModel semanticModel)
+	private static bool IsFluxorComponentBase(INamedTypeSymbol symbol, INamedTypeSymbol? fluxorComponent, INamedTypeSymbol? fluxorLayout)
+		=> DerivesFrom(symbol, fluxorComponent) || DerivesFrom(symbol, fluxorLayout);
+
+	private static bool DerivesFrom(INamedTypeSymbol symbol, INamedTypeSymbol? candidateBaseType)
 	{
-		if (
-			invocation.Expression is MemberAccessExpressionSyntax memberAccess
-			&& memberAccess.Expression is BaseExpressionSyntax)
+		var baseType = symbol.BaseType;
```

Comment:
May be worth leaving a comment for the future that this helper doesn't support generic base types correctly?